specs/execution-api.md

```diff
@@ -1,136 +1,154 @@
-the state machine via `GetCommitmentState`. If started on a fresh rollup
-these will all be the same block. If running against a state machine with
-previous block data, Conductor must also track the block hash of any blocks
-between commitments, it will call `BatchGetBlocks` to get block information
-between commitments.
+Upon startup, conductor starts its first execution session by calling `CreateExecutionSession`.
+This returns an `ExecutionSession`, containing the necessary information for the
+client to drive execution for the duration of the session.
 
 ### Execution & Commitments
 
-From the perspective of the sequencer:
-
-- `SOFT` commitments have been fully committed to by the sequencer consensus.
-- `FIRM` commitment indicates that the block has been written and has been
-  propagated across the DA network.
+From the perspective of the Conductor:
 
-When configuring conductor, you can configure the time at which blocks are
-executed in your rollup using the `execution_commitment_level` in the config
-file. If this is configured to a higher level of commitment, no action will be
-taken upon receiving lower commitments.
+- `Soft` commitments have been fully committed by the Sequencer consensus.
+- A `Firm` commitment indicates that the block has been written and propagated
+  across the DA network.
 
-`ExecuteBlock` is called to create a new rollup block when the
-`execution_commitment_level` has been reached for a given block. Upon receipt of
-a new block, the conductor calls `UpdateCommitmentState` to update the
-commitment at the level of the `execution_commitment_level` and any level above
-it.
+When configuring Conductor, the threshold at which blocks are executed on the rollup
+can be set via the `execution_commitment_level` in the config file. `ExecuteBlock`
+is called to create a new rollup block when the `execution_commitment_level` has
+been reached for a given block. Upon receipt of the executed block, Conductor calls
+`UpdateCommitmentState` to update the commitment at the level of the
+`execution_commitment_level` and any level above it.
 
-`execution_commitment_level` options, and changes to execution:
+`execution_commitment_level` options and changes to execution:
 
-- `SOFT`
-  - upon receiving a new sequencer block N from sequencer:
+- `SoftOnly`
+  - upon receiving a new sequencer block N from the Sequencer:
     - `ExecuteBlock` will be called with data from the sequencer block N, then
-    - `UpdateCommitmentState` will be called again to update the `SAFE` to N
+    - `UpdateCommitmentState` will be called to update the `soft` block to N
+- `SoftAndFirm`
+  - upon receiving a new sequencer block N from the Sequencer:
+    - `ExecuteBlock` will be called with data from the sequencer block N, then
+    - `UpdateCommitmentState` will be called to update the `soft` block to N
   - upon reading new blocks from DA containing all of blocks K->N, where K is
```

Comment:
We should talk about Celestia here. We don't use `DA` anywhere else, and we also don't have any alternatives.

```suggestion
- upon reading new blocks from Celestia containing all of blocks K->N, where K is
```